build/protobuf.mk

```diff
@@ -123,9 +123,9 @@ $(CPP_HEADERS) $(CPP_SOURCES): $(PROTOC) $(CPP_PROTOS)
 	$(SED_INPLACE) -E '/gogoproto/d' $(CPP_HEADERS) $(CPP_SOURCES)
 	@# For c++, protoc generates a directory structure mirroring the package
 	@# structure (and these directories must be in the include path), but cgo can
-	@# only compile a single directory so we symlink the generated pb.cc files
+	@# only compile a single directory so we "symlink" the generated pb.cc files
 	@# into the storage/engine directory.
 	@# We use `find` and not `git ls-files` here because `git ls-files` will
-	@# include deleted files (i.e. these very symlinks) in its output, resulting
-	@# in recursive symlinks, which is Bad™.
-	(cd $(NATIVE_ROOT) && find . -name *.pb.cc | xargs -I % ln -sf % .)
+	@# include deleted files (i.e. these very "symlinks") in its output, resulting
+	@# in recursive "symlinks", which is Bad™.
+	(cd $(NATIVE_ROOT) && find . -name *.pb.cc | sed 's!./!!' | xargs -I % sh -c 'echo "#include \"%\"" > $$(echo % | tr / _)')
```

Comment:
Would be nice to add a small comment here that describes what the sed, xargs and echo are doing.